zap/src/main/java/org/zaproxy/zap/core/scanner/InputVectorBuilder.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.core.scanner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.zaproxy.zap.core.scanner.InputVector.PayloadFormat;
+
+/**
+ * {@code InputVectorBuilder} provides an easy way to build {@code InputVectors} and validates the
+ * provided inputs.
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class InputVectorBuilder {
+
+    private Map<Integer, InputVector> inputVectorsMap = new HashMap<>();
+
+    /**
+     * Note: calling this method with same NameValuePair will overwrite the earlier name and value
+     * with the newer one.
+     *
+     * @param nameValuePair
+     * @param param
+     * @param value
+     * @param namePayloadFormat
+     * @param valuePayloadFormat
+     * @return {@code InputVectorBuilder}
+     */
+    public InputVectorBuilder setNameAndValue(
+            NameValuePair nameValuePair,
+            String param,
+            String value,
+            PayloadFormat namePayloadFormat,
+            PayloadFormat valuePayloadFormat) {
+        inputVectorsMap.put(
+                nameValuePair.getPosition(),
+                new InputVector(
+                        nameValuePair.getPosition(),
+                        param,
+                        value,
+                        namePayloadFormat,
+                        valuePayloadFormat));
+        return this;
+    }
+
+    /**
+     * Note: calling this method with same NameValuePair will overwrite the earlier value with the
+     * newer one.
+     *
+     * @param nameValuePair
+     * @param value
+     * @param valuePayloadFormat
+     * @return {@code InputVectorBuilder}
+     */
+    public InputVectorBuilder setValue(
+            NameValuePair nameValuePair, String value, PayloadFormat valuePayloadFormat) {
+        return this.setNameAndValue(
+                nameValuePair,
+                nameValuePair.getName(),
+                value,
+                PayloadFormat.ALREADY_ESCAPED,
```

Comment:
This needs to be reviewed, I think some of the variants create the `NameValuePair`s with unescaped name/value.

Reply:
Does it matters? I think we can go with AlreadyEscaped parameter as default as this Api is not used till now.

Reply:
It does because it might change the meaning of the name/value, but not a "blocker" (we can tweak as needed later).